pkg/controller/manager/status.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2024 ScyllaDB
+
+package manager
+
+import (
+	"context"
+
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	"github.com/scylladb/scylla-operator/pkg/helpers/slices"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+func (c *Controller) calculateStatus(sc *scyllav1.ScyllaCluster, managerState *state) *scyllav1.ScyllaClusterStatus {
+	status := sc.Status.DeepCopy()
+
+	status.Repairs = []scyllav1.RepairTaskStatus{}
+	for _, rt := range sc.Spec.Repairs {
+		managerTaskStatus, ok := managerState.RepairTasks[rt.Name]
+		if !ok {
+			continue
+		}
+		status.Repairs = append(status.Repairs, scyllav1.RepairTaskStatus(managerTaskStatus))
+	}
+
+	status.Backups = []scyllav1.BackupTaskStatus{}
+	for _, bt := range sc.Spec.Backups {
+		managerTaskStatus, ok := managerState.BackupTasks[bt.Name]
+		if !ok {
+			continue
+		}
+		status.Backups = append(status.Backups, scyllav1.BackupTaskStatus(managerTaskStatus))
+	}
+
+	return status
+}
+
+func syncStatusWithManagerState(sc *scyllav1.ScyllaCluster, status *scyllav1.ScyllaClusterStatus, managerState *state) {
```

Comment:
>  wouldn't sync it because tasks which are not found in managerState would stay in status.

hmm that would only happen if a task, which was in spec, was removed from manager state between first calling calculate status and syncing it again, but you're right, there's an edge case which it doesn't cover. Do you have any suggestions for improving that, while still taking client errors into account, without too much refactoring of the actions? Imo the best approach would be to collect action errors (specifically task-related action errors, not the cluster-related ones) and passing them to this function, instead of modifying the status on action execution, but that would require changing the existing code to a bigger extent than now.

Reply:
> Edit: implemented this for now

updated the implementation since I introduced a bug there